### ISSUE TITLE
custom obj function and eval function

### ISSUE THREAD

**Issue by IversonR** [dmlc/xgboost#2534]
In the newest release, I used the custom obj and eval function, in the process of running, the 'silent' that includes original eval and my custom eval. This is why?

**Comment by hcho3** [comment]
Can you upload your configuration file, so that we can re-produce the issue? Thanks!

**Comment by IversonR** [comment]
[xgb_scale.pdf](https://github.com/dmlc/xgboost/files/1167997/xgb_scale.pdf)

**Comment by hcho3** [comment]
Could you upload your code in text file?

**Comment by IversonR** [comment]
This is text file, thank you for your help！


[xgb_scale.txt](https://github.com/dmlc/xgboost/files/1168002/xgb_scale.txt)

**Comment by hcho3** [comment]
@IversonR I've taken a look at this issue. XGBoost appears to be working as designed: if you don't specify the parameter `eval_metric`, XGBoost will print RMSE by default. This is the case even when the custom loss function is supplied. Hope it helps!

**Comment by IversonR** [comment]
Thanks for your reply, I gave your text file which is specified the parameter about eval_metric, last time. Did you look the file? Or there are some questions about specified the parameter. I am looking forward to your suggestion. Thanks!
